src/stardict_lib.cpp

```diff
@@ -831,23 +831,30 @@ bool WordListIndex::lookup(const char *str, glong &idx)
 
 bool SynFile::load(const std::string &url, gulong wc)
 {
+
     struct stat stat_buf;
     if (!stat(url.c_str(), &stat_buf)) {
+
         MapFile syn;
         if (!syn.open(url.c_str(), stat_buf.st_size))
             return false;
-        const gchar *current = syn.begin();
+
+        syndatabuf = (gchar *)g_malloc(stat_buf.st_size);
```

Comment:
You can use MapFile by itself without allocating memory

Reply:
Okay, now it keeps MapFile around instead of copying it to a buffer.